design/upf-selection-design.md

```diff
@@ -0,0 +1,185 @@
+    imsis:
+      - "001010000000001"
+      - "001010000000002"
+      - "001010000000003"
+      - "001010000000005"
+      - "001010000000008"
+      - "001010000000009"
+    ip-domain-name: "pool1"
+    ip-domain-expanded:
+      dnn: internet
+      dns-primary: "10.176.0.11"        # Value sent to UE
+      mtu: 1460                        # Value sent to UE when PDU Session Established
+      ue-ip-pool: "172.250.1.0/16"     # IP address pool for subscribers
+      ue-dnn-qos:
+        dnn-mbr-downlink: 1000         # UE level downlink QoS (Maximum bit rate per UE)
+        dnn-mbr-uplink: 1000           # UE level uplink QoS (Maximum bit rate per UE)
+        bitrate-unit: Mbps             # Unit for above QoS rates
+        traffic-class:                 # Default bearer QCI/ARP (not used in 5G)
+          name: "platinum"
+          qci: 9
+          arp: 6
+          pdb: 300
+          pelr: 6
+    site-info: "enterprise"
+  - name: "5g-gnbsim-user-group2"
+    imsis:
+      - "001010000000001"
+      - "001010000000002"
+      - "001010000000003"
+      - "001010000000005"
+      - "001010000000008"
+      - "001010000000009"
+    ip-domain-name: "pool1"
+    ip-domain-expanded:
+      dnn: sdcore
+      dns-primary: "10.176.0.11"        # Value sent to UE
+      mtu: 1460                        # Value sent to UE when PDU Session Established
+      ue-ip-pool: "172.252.1.0/16"     # IP address pool for subscribers
+      ue-dnn-qos:
+        dnn-mbr-downlink: 1000         # UE level downlink QoS (Maximum bit rate per UE)
+        dnn-mbr-uplink: 1000           # UE level uplink QoS (Maximum bit rate per UE)
+        bitrate-unit: Mbps             # Unit for above QoS rates
+        traffic-class:                 # Default bearer QCI/ARP (not used in 5G)
+          name: "platinum"
+          qci: 9
+          arp: 6
+          pdb: 300
+          pelr: 6
+    site-info: "enterprise"
+ network-slices:
+            - name: "default"      # can be any unique slice name
+              slice-id:            # must match with slice configured in gNB, UE
+                sd: "000000"
+                sst: 1
+              site-device-group:
+              - "5g-gnbsim-user-group1"   # All UEs in this device-group are assigned to this slice
+              # Applicaiton filters control what each user can access.
+              # Default, allow access to all applications
+              application-filtering-rules:
+              - rule-name: "ALLOW-ALL"
+                priority: 250
+                action: "permit"
+                endpoint: "0.0.0.0/0"
+              site-info:
+                # Provide gNBs and UPF details and also PLMN for the site
+                gNodeBs:
+                - name: "gnb1"
+                  tac: 1
+                - name: "gnb2"
+                  tac: 2
+                plmn:
+                  mcc: "001"
+                  mnc: "01"
+                site-name: "enterprise"
+                upf:
```

Comment:
This is where it limits one UPF per Slice.. We need changes in this config so that it remains backward compatible.

Reply:
The upf field has been replaced by upfs to accommodate multiple UPF definitions. Each UPF is specified as an item in a list, with its own upf-name and upf-port